### PR TITLE
Fix ClientSession.close() was never awaited

### PR DESCRIPTION
**Pull request by ciscorn**
aiohttp's `ClientSession.close()` has been changed to a coroutine since version 2.??. (see #22)

I think we need a custom connection pool to override the sync close method.

closes #22 #43

### CODE REVIEW

**Comment by rciorba** [comment]
> I think we have to make a custom connection pool to override the sync close method.

There's still a problem. When using only one connection, you'll get the DummyConnectionPool, which doesn't await.

I tried something different on my fork, but aren't yet happy about it. Not sure how idiomatic it is to return a list of futures from close (I'm new to async), but this seems to work with both sync and async usage:
https://github.com/elastic/elasticsearch-py/compare/master...rciorba:await_friendly_close?expand=1

**Comment by ciscorn** [comment]
@rclorba Thanks. I updated and fixed my PR.

**Comment by hh-h** [comment]
Guys, could someone review, merge and release this? Thanks!

**Comment by fxdgear** [comment]
Appreciate the PR @ciscorn. Would be helpful if you could include some testing around this as well. 

Thanks

File: test_elasticsearch_async/conftest.py [hunk 1]
```diff
@@ -61,17 +62,19 @@ def port():
...
-    server = DummyElasticsearch(debug=True, keep_alive=75)
+    server = DummyElasticsearch(debug=True, loop=event_loop)
     f = event_loop.create_server(server, '127.0.0.1', port)
     event_loop.run_until_complete(f)
     yield server
     event_loop.run_until_complete(server.shutdown(timeout=.5))
 
 @yield_fixture
 def client(event_loop, server, port):
+    logger = logging.getLogger('elasticsearch')
```
Comment (ciscorn) [inline test_elasticsearch_async/conftest.py]: required for the testcase `test_elasticsearch_async/test_connection.py::test_request_is_properly_logged`

File: test_elasticsearch_async/conftest.py [hunk 2]
```diff
@@ -61,17 +62,19 @@ def port():
...
+    logger.setLevel(logging.DEBUG)
```
Comment (ciscorn) [inline test_elasticsearch_async/conftest.py]: required for the testcase `test_elasticsearch_async/test_connection.py::test_request_is_properly_logged`

**Comment by ciscorn** [comment]
@fxdgear 
I wrote some tests around this.
(BTW: Some part of the test suite was broken, so I also fixed them.)
All tests pass on my environment.

**Comment by yarara** [comment]
Hello, please review this PR and merge it!!! Thanks

**Pull request merged**